api/types/oidc.go

```diff
@@ -446,6 +447,15 @@ func (o *OIDCConnectorV3) CheckAndSetDefaults() error {
 		}
 	}
 
+	if o.Spec.ClientRedirectSettings != nil {
+		for _, cidrStr := range o.Spec.ClientRedirectSettings.InsecureAllowedCidrRanges {
+			_, err := netip.ParsePrefix(cidrStr)
+			if err != nil {
+				return trace.BadParameter("bad CIDR range in insecure_allowed_cidr_ranges '%s': %v", cidrStr, err)
```

Comment:
This (and to a much lesser extent the later check when using the string) makes me uneasy - as written, a connector that becomes broken (say, because the rules for `netip.ParsePrefix` are tightened due to some bug) will prevent all logins instead of failing to allow the specific range that has become broken, and preventing SSO logins can be very problematic if break glass access wasn't planned.

Reply:
That's a good point. I moved the CIDR validation to when the OIDC connector resources are getting created or updated to prevent existing connector resources from being used in the future due to a `netip` bug